extprocconfig/extprocconfig.go

```diff
@@ -65,6 +73,18 @@ type Config struct {
 	Rules []RouteRule `yaml:"rules"`
 }
 
+// TokenUsageMetadata is the namespace and key to be used in the filter metadata to store the usage token.
```

Comment:
As https://github.com/envoyproxy/envoy/pull/37567 merged, we can support this with `%DYNAMIC_METADATA(NAMESPACE:KEY*):Z%`, is that right?

Reply:
exactly!

Reply:
this is so cool right - we can remove the whole bunch of hacky code from MVP

Reply:
Should we only support metadata? it's fine for now, but it's worth thinking more.

Reply:
yeah i guess? token extraction happens *after* response headers are sent to downstream, so we cannot set it in response headers. That makes metadata is the only position to store this.